libqtile/widget/notify.py

```diff
@@ -98,21 +100,32 @@ def real_update(self, notif):
         self.set_notif_text(notif)
         self.current_id = notif.id - 1
         if notif.timeout and notif.timeout > 0:
-            self.timeout_add(notif.timeout / 1000, self.clear)
+            self.timeout_add(
+                notif.timeout / 1000, self.clear, method_args=(ClosedReason.expired,)
+            )
         elif self.default_timeout:
-            self.timeout_add(self.default_timeout, self.clear)
+            self.timeout_add(
+                self.default_timeout, self.clear, method_args=(ClosedReason.expired,)
+            )
         self.bar.draw()
         return True
 
     def display(self):
         self.set_notif_text(notifier.notifications[self.current_id])
         self.bar.draw()
 
-    def clear(self):
+    def clear(self, reason=ClosedReason.dismissed):
+        notifier._service.NotificationClosed(self.current_id, reason)
```

Comment:
This doesn't actually remove the notification from the widget does it? It looks like it just hides the text and moves the display to the previous notification. Presumably we could make it re-appear by scrolling back and forwards again.

Should a dismissed notification be removed entirely?

Couldn't we also end up with an `IndexError` if we

Reply:
I think it should be considered a 'dismiss' rather than a 'delete'. It sets the current ID to the final one, not to the previous.